### PR TITLE
Initial QuadKey implementation

### PR DESCRIPTION
**Pull request by joewalsh**

### CODE REVIEW

File: Wikipedia/Code/QuadKey.swift [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+            let shiftedLongitudePart = quadKeyLongitudePart >> i
+            quadKey = (quadKey << 1) | (shiftedLongitudePart & 1)
+            let shiftedLatitudePart = quadKeyLatitudePart >> i
+            quadKey = (quadKey << 1) | (shiftedLatitudePart & 1)
+            i -= 1
+        }
+        self.init(quadKey)
+    }
+    
+    func adjusted(downBy precisions: QuadKeyPrecision) -> QuadKey {
```
Comment (montehurd) [inline Wikipedia/Code/QuadKey.swift]: Should `precisions` be `precision`?

File: WikipediaUnitTests/Code/QuadKeyTests.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                    let fullPrecisionQuadKey = QuadKey(latitude: lat, longitude: lon)
+                    let adjustedQuadKey = fullPrecisionQuadKey.adjusted(downBy: 32 - p)
+                    XCTAssertEqual(quadKey, adjustedQuadKey, "QuadKeys should match no matter how they were calculated\n\(adjustedQuadKey.bitmaskString)\n\(quadKey.bitmaskString)\n\(fullPrecisionQuadKey.bitmaskString)\n\n\(fullPrecisionLatPart.bitmaskString)\n\(latPart.bitmaskString)\n\n\(fullPrecisionLonPart.bitmaskString)\n\(lonPart.bitmaskString)")
+                    let coord = QuadKeyCoordinate(quadKey: quadKey, precision: p)
+                    XCTAssertEqual(coord.latitudePart, latPart, "latPart: \(latPart) != \(coord.latitudePart)")
+                    XCTAssertEqual(coord.longitudePart, lonPart, "lonPart: \(lonPart) != \(coord.longitudePart)")
+                    let calculatedLat = QuadKeyDegrees(latitudePart: coord.latitudePart, precision: p)
+                    let calculatedLon = QuadKeyDegrees(longitudePart: coord.longitudePart, precision: p)
+                    XCTAssertEqualWithAccuracy(calculatedLat, lat, accuracy: p.deltaLatitude, "latitude is wrong at poom level \(p)")
```
Comment (montehurd) [inline WikipediaUnitTests/Code/QuadKeyTests.swift]: What's `poom level`?

Comment (joewalsh) [inline WikipediaUnitTests/Code/QuadKeyTests.swift]: used to be zoom level, should be precision

File: Wikipedia/Code/QuadKey.swift (hunk 1 above)
Comment (joewalsh) [inline Wikipedia/Code/QuadKey.swift]: yes

File: WikipediaUnitTests/Code/QuadKeyTests.swift (hunk 2 above)
Comment (montehurd) [inline WikipediaUnitTests/Code/QuadKeyTests.swift]: ah gotcha

**Pull request merged**